lib/Game_repository.rb

```diff
@@ -0,0 +1,31 @@
+class GameRepository
+  def initialize
+    @rolls = []
+  end
+  
+  def add(roll)
+   @rolls << roll
+  end
+  
+  def score_card
+    #@rolls.sum
+  
+    roll_index = 0
+    score = 0
+     10.times do
+     if @rolls[roll_index] + @rolls[roll_index + 1] == 10 #spare
+      score += @rolls[roll_index] + @rolls[roll_index + 1] + @rolls[roll_index + 2]
+      elsif @rolls[roll_index] +@rolls[roll_index +1] != 10 # frame score accumulation <10
+      score += @rolls[roll_index] + @rolls[roll_index + 1] 
+      elsif @rolls[roll_index] == 10 # Check Strike- previous  fail appeared to be due to rspec - further tests needed to ensure- bonus score is calculated the bonus feature.
+      score += @rolls[roll_index] + @rolls[roll_index + 1] + @rolls[roll_index + 2]
+      roll_index += 1
+   end
+      roll_index += 2
+     end
+      score
+    end
```

Comment:
This could be refactored to improve readability by creating descriptive variable names and extracting some logic into separate functions. For example:

```rb
def score_card
  roll_index = 0
  score = 0
  
  10.times do
    roll = @rolls[roll_index]
    next_roll = @rolls[roll_index + 1]
    if spare?(roll, next_roll)
      subsequent_roll = @rolls[roll_index + 2]
      score += roll + next_roll + subsequent_roll
    elsif !spare(roll, next_roll)
      # etc...
    end
  end
end
```

Reply:
Additionally, because the first `if` and `elsif` statements cover all possible outcomes (Do the current roll and the next roll add up to ten?) the later `elsif` will never be run.